java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -78,13 +78,16 @@ public Scope createStaticStarImportScope(JavaSymbol owner) {
   }
 
   public JavaType resolveTypeSubstitution(JavaType type, JavaType definition) {
-    if(isParametrizedType(definition)) {
+    if (isParametrizedType(definition)) {
       return substituteTypeParameter(type, ((JavaType.ParametrizedTypeJavaType) definition).typeSubstitution);
     }
     return type;
   }
 
   private JavaType substituteTypeParameter(JavaType type, TypeSubstitution substitution) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "substituteTypeParameter" is 15 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)